qstack/spahm/rho/utils.py

```diff
@@ -70,33 +74,77 @@ def get_xyzlist(xyzlistfile):
     return np.loadtxt(xyzlistfile, dtype=str, ndmin=1)
 
 
-def load_reps(f_in, from_list=True, single=True, with_labels=False, local=True, reaction=False):
+def load_reps(f_in, from_list=True, srcdir=None, single=False, with_labels=False, local=True, sum_local=False, printlevel=0, progress=False):
+    '''
+    A function to load representations from txt-list/npy files.
+        Args:
+            - f_in: the name of the input file
+            - from_list(bool): if the input file is a txt-file containing a list of paths to the representations
+            - single: if the input file is .npy AND contains a single representation (not an array of representation)
+            - with_label(bool): saves a list of tuple (filename, representation)
+            - local(bool): if the representations is local
+            - sum_local(bool): if local=True then sums the local components
+            - printlevel(int): level of verbosity
+        Return:
+            np.array with shape (N,M) where N number of representations M dimmensionality of the representation
+            OR tuple (N,np.array(N,M)) containing filenames in pos 0
+    '''
+    if srcdir == None:
+        path2list = os.getcwd()
+    else:
+        path2list = srcdir
     if from_list:
         X_list = get_xyzlist(f_in)
-        Xs = [np.load(f_X, allow_pickle=True) for f_X in X_list]
+        Xs = []     # Xs must be a list of representations (local or global) whose len() = # of representations
+        for f_X in X_list:
+            Xs.append(np.load(os.path.join(path2list,f_X), allow_pickle=True))
+    elif single:
+        Xs = [np.load(f_in, allow_pickle=True)] # if the given file contains a single representation create a one-element list
     else:
-        Xs = [np.load(f_in, allow_pickle=True)]
```

Comment:
here I couldn't use `ndmin` because at this point we don't yet if the representation-files contain labels or not, which would change the min. dim. requested.

So I made it a list (everywhere) of one-element and added a comment!

_let me know if you have a better/shorter idea_

Reply:
I think it maybe makes more sense to check if the first element contains the label. Not sure that asking user is it's a single/multiple representation is a good idea, may lead to more errors.

Reply:
And I think it should print what type of representation storage is detected!